inf-clojure.el

```diff
@@ -293,7 +293,10 @@ See http://blog.jorgenschaefer.de/2014/05/race-conditions-in-emacs-process-filte
   "Set the REPL type if has not already been set.
 It requires a REPL PROC for inspecting the correct type."
   (if (not inf-clojure-repl-type)
-      (setq inf-clojure-repl-type (inf-clojure--detect-repl-type proc))
+      (let ((detected-type (inf-clojure--detect-repl-type proc)))
+        (with-current-buffer inf-clojure-buffer
+          (setq inf-clojure-repl-type detected-type))
+        (setq inf-clojure-repl-type detected-type))
```

Comment:
Why do we need to set the REPL type in the source buffer? We can always infer this from the REPL buffer for the project.

Reply:
Uhm are there any implications in terms of performance in getting the buffer and from it the value for every operation? Remember this is done on each key press (for completion). If not I can change things, but this patch was definitely easier.

I think also that removing the buffer local would work but I was thinking that at some point we should have more than one repl (per project) so I don't know how it work then.